src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -635,64 +635,76 @@ bool GetPeerNodeInfo(CommandHandler * commandHandler, const ConcreteCommandPath
-                    if (CHIP_NO_ERROR != err) {
-                        LogErrorOnFailure(err);
-                        if (err == CHIP_ERROR_BUSY) {
-                            Commands::QueryImageResponse::Type busyResponse;
-                            busyResponse.status = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusBusy);
-                            busyResponse.delayedActionTime.SetValue(response.delayedActionTime.ValueOr(kDelayedActionTimeSeconds));
-                            handler->AddResponse(cachedCommandPath, busyResponse);
-                            handle.Release();
-                            return;
-                        }
-                        handler->AddStatus(cachedCommandPath, Protocols::InteractionModel::Status::Failure);
-                        handle.Release();
-                        gOtaSender.ResetState();
-                        return;
-                    }
-                    auto targetNodeId
-                        = handler->GetExchangeContext()->GetSessionHandle()->AsSecureSession()->GetLocalScopedNodeId();
+                // The logic we are following here is if the protocols send by the requestor are not supported, we can't transfer
+                // the image even if we had an image available and we would return a Protocol Not Supported status. Assumption here
+                // is the requestor would send all the supported protocols. If one/more of the protocols send by the requestor are
+                // supported, we check if an image is not available due to various reasons - image not available, delegate reporting
+                // busy, we will respond with the status in the delegate response. If update is available, we try to prepare for
+                // transfer and build the uri in the response with a status of Image Available
+
+                // If the protocol requested is not supported, return status - Protocol Not Supported
+                if (!isBDXProtocolSupported) {
+                    Commands::QueryImageResponse::Type response;
+                    response.status
+                        = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusDownloadProtocolNotSupported);
+                    handler->AddResponse(cachedCommandPath, response);
+                    handle.Release();
+                    return;
+                }
 
-                    char uriBuffer[kMaxBDXURILen];
-                    MutableCharSpan uri(uriBuffer);
-                    err = bdx::MakeURI(targetNodeId.GetNodeId(), AsCharSpan(data.imageURI), uri);
-                    if (CHIP_NO_ERROR != err) {
-                        LogErrorOnFailure(err);
-                        handler->AddStatus(cachedCommandPath, Protocols::InteractionModel::Status::Failure);
+                Commands::QueryImageResponse::Type delegateResponse;
+                ConvertFromQueryImageResponseParms(data, delegateResponse);
```

Comment:
Followup: we should fix the naming of that function: s/Parms/Params/.

Reply:
good catch. will do